### ISSUE TITLE
I found out how to solve the Windows 10 2004+ (20H1) version of the high precision timer failure and the solution

### ISSUE THREAD

**Issue by cyz0054** [amitxv/TimerResolution#7]
Hello, a great god, and I find a solution to the Windows 2004 + version (h1) 20 high precision timer failure and its solution to the original address: https://blog.csdn.net/bluesen/article/details/115404636

If you can't see this link I'll copy and paste the content here

High-precision timers are used in many places, such as real-time communication, multimedia processing, and game development.

I developed a softswitch system, multimedia processing to use high-precision timers, in windows common practice is:

Start by setting the timer to 1 millisecond:

timeBeginPeriod(1);

This is crucial because otherwise the timer will have a very low precision of about 15.6 milliseconds and will be practically unusable.

Then use a set of functions like this:

CreateTimerQueue();

CreateTimerQueueTimer(&t, timer, (WAITORTIMERCALLBACK)OneMilliSecondProc, this, 0, 10, 0);   // Call the callback every 10ms

However, in the latest version of windows10, the precision is set to 1 millisecond, and the program does not error, but the actual precision is reduced to 15.6 milliseconds. In fact, the precision setting does not work at all.

The solution is to wrap your own timer using the C++1x time function, as shown in the following example:

// Timer function

int Timer11()

{

chrono::steady_clock::time_point tp = std::chrono::steady_clock::now();

auto dt_micri = std::chrono::microseconds(10000);   // 10ms

while (true) {

tp += dt_micri;

this_thread::sleep_until(tp);

// callback function, application is written in this function

OneMilliSecondProc();

}

return(0);

}

// Example call

thread thd(Timer11);   // Create thread

thd.detach();

My compiler is Microsoft Visual C++ 2017. After testing, it is found that the precision of the timer made by this method is quite high, and the code is simple and universal. It works normally and runs well under the latest version of windows10.

Can you write a program similar to SetTimerResolution.exe based on this solution

**Comment by amitxv** [comment]
Hi, this is an alternative for a sleep timer and has nothing to do with raising the clock interrupt frequency.
